chalicelib/utils.py

```diff
@@ -58,21 +59,22 @@ def workspace_exists(self):
             s.mount('https://', HTTPAdapter(
                 max_retries=Retry(total=4,
                                   backoff_factor=1,
-                                  redirect=False,
-                                  history=(),
-                                  raise_on_status=True,
+                                  raise_on_status=False,
                                   status_forcelist=[401, 500, 502, 503, 504])
             ))
-            return s.get(url=url, headers=headers)
-        except:
-            return Response(body={},
-                            headers={'Content-Type': 'application/json'},
-                            status_code=500)
+            resp = s.get(url=url, headers=headers)
+            resp.raise_for_status()
+            return resp
+        except HTTPError as err:
```

Comment:
See previous comment; you don't need a `try` -- you are doing a throw, only to catch it, and then return it. You can save a step by just returning it.

Reply:
re. try, I removed it. strange though, I seemed to have been convinced that we need it. re. `raise_for_status`, it's actually slightly better to leave without (but it was required for try/except) as it will raise an exception for all 4xx/5xx. and thus would raise one if the workspace doesn't exist (404).